src/server/routes/api/v1/users.ts

```diff
@@ -16,16 +17,18 @@ export default function (
     async (
       request: FastifyRequest<{
         Body: {
-          firstName: string;
-          lastName: string;
           email: string;
           password: string;
+          role: Role;
         };
       }>,
       reply: FastifyReply,
     ) => {
       try {
-        const user = await fastify.services.user.create(request.body);
+        const user = await fastify.services.user.create(
+          request.body,
+          fastify.user.email,
+        );
```

Comment:
Looks like we want to provide the current user here, not only the email.

I suggest fetching the "current user" from database and pass it to this function instead of using the token claims.

I would change this to be:

```suggestion
// fetch request user from database
const currentUser = await fastify.services.user.findByEmail(fastify.user.email);

// I'm not checking if the user exists because it comes from a token signed by this server
// so the user should exist.
// This is valid unless a User deletion is introduced

// Then use the user on its last state to proceed with this request
const user = await fastify.services.user.create(
  currentUser,
  request.body, // this is the "CreateUserDto"
);
```